### ISSUE TITLE
[dart2wasm] Unboxed Wasm types are considered as dynamic call receivers

### ISSUE THREAD

**Issue by osa1** [dart-lang/sdk#50314]
I debugged this issue at least twice in different CLs. I want to report and fix it before I forget about it again.

This program works compiles and runs as expected:

```dart
void main() {
  print((null as dynamic).toInt());
}
```

Now, if I make `WasmI32.toInt` method reachable by introducing a reference to it:

```dart
import 'dart:wasm';

void main() {
  WasmI64.fromInt(123).toInt();
  print((null as dynamic).toInt());
}
```

This fails to compile:

```
Unhandled exception:
Wasm numeric types can't be nullable
```

The problem seems to be that the dynamic dispatcher considers `WasmI64.toInt` a possible target for the dynamic `toInt` call, but that method is special in that its receiver is an unboxed Wasm `I64` value.

Other classes in wasm_types library are also similar.

I think we should filter those out when generating method searches in dynamic calls.

@askeksa-google any thoughts on this?

An example of how this happens in unrelated CLs is in https://dart-review.googlesource.com/c/sdk/+/266020 we use `WasmI64.toInt` in `/~` implementation. So `(null as dynamic).toInt()` causes a compile-time panic if the program uses `/~` elsewhere.

These two tests also fail because of this issue:

- language/variable/inference_captured_variable_test
- lib/typed_data/typed_data_list_test

**Comment by askeksa-google** [comment]
Yes, indeed it makes sense to exclude all members declared in any subtype of `_WasmBase` from the dynamic targets.

**Comment by osa1** [comment]
The commit message doesn't say what it's fixing so I don't know if this was intentional or not, but 16598688b898f40fd83740f4342167e2dd6c78fa fixed this crash.

**Comment by osa1** [comment]
https://dart-review.git.corp.google.com/c/sdk/+/268761

**Comment by osa1** [comment]
Fixed with 5eb0a65d90a5890c8ed5e49a7c38691f1ef5f45d.
